WelsonJS.Toolkit/EsentInterop/RetrieveColumnHelpers.cs

```diff
@@ -0,0 +1,1172 @@
+        {
+            // We cannot support this request when there is no way to indicate that a column reference is returned.
+            if ((grbit & (RetrieveColumnGrbit)0x00020000) != 0)  // UnpublishedGrbits.RetrieveAsRefIfNotInRecord
+            {
+                throw new EsentInvalidGrbitException();
+            }
+
+            unsafe
+            {
+                const int DataSize = sizeof(ulong);
+                ulong data;
+                var pointer = new IntPtr(&data);
+                int actualDataSize;
+                JET_wrn wrn = JetRetrieveColumn(
+                    sesid, tableid, columnid, pointer, DataSize, out actualDataSize, grbit, null);
+                return CreateReturnValue(data, DataSize, wrn, actualDataSize);
+            }
+        }
+
+#if MANAGEDESENT_SUPPORTS_SERIALIZATION
+        /// <summary>
+        /// Deserialize an object from a column.
+        /// </summary>
+        /// <param name="sesid">The session to use.</param>
+        /// <param name="tableid">The table to read from.</param>
+        /// <param name="columnid">The column to read from.</param>
+        /// <returns>The deserialized object. Null if the column was null.</returns>
+        public static object DeserializeObjectFromColumn(JET_SESID sesid, JET_TABLEID tableid, JET_COLUMNID columnid)
+        {
+            return DeserializeObjectFromColumn(sesid, tableid, columnid, RetrieveColumnGrbit.None);
+        }
+
+        /// <summary>
+        /// Deserialize an object from a column.
+        /// </summary>
+        /// <param name="sesid">The session to use.</param>
+        /// <param name="tableid">The table to read from.</param>
+        /// <param name="columnid">The column to read from.</param>
+        /// <param name="grbit">The retrieval options to use.</param>
+        /// <returns>The deserialized object. Null if the column was null.</returns>
+        [SuppressMessage("Exchange.Security", "EX0043:DoNotUseBinarySoapFormatter", Justification = "Suppress warning in current code base.The usage has already been verified.")]
+        public static object DeserializeObjectFromColumn(JET_SESID sesid, JET_TABLEID tableid, JET_COLUMNID columnid, RetrieveColumnGrbit grbit)
+        {
+            // We cannot support this request when there is no way to indicate that a column reference is returned.
+            if ((grbit & (RetrieveColumnGrbit)0x00020000) != 0)  // UnpublishedGrbits.RetrieveAsRefIfNotInRecord
+            {
+                throw new EsentInvalidGrbitException();
+            }
+
+            int actualSize;
+            if (JET_wrn.ColumnNull == Api.JetRetrieveColumn(sesid, tableid, columnid, null, 0, out actualSize, grbit, null))
+            {
+                return null;
+            }
+
+            using (var stream = new ColumnStream(sesid, tableid, columnid))
+            {
+                var deseriaizer = new BinaryFormatter();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The BinaryFormatter type is dangerous and is not recommended for data processing. Applications should stop using BinaryFormatter as soon as possible, even if they believe the data they are processing to be trustworthy. BinaryFormatter is insecure and can not be made secure. Learn more - https://learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-security-guide